core/src/main/java/io/questdb/griffin/engine/functions/date/GenerateSeriesTimestampStringRecordCursorFactory.java

```diff
@@ -0,0 +1,285 @@
+    private GenerateSeriesTimestampStringRecordCursor cursor;
+
+    public GenerateSeriesTimestampStringRecordCursorFactory(Function startFunc, Function endFunc, Function stepFunc, int position) throws SqlException {
+        super(METADATA, startFunc, endFunc, stepFunc, position);
+    }
+
+    @Override
+    public RecordCursor getCursor(SqlExecutionContext executionContext) throws SqlException {
+        if (cursor == null) {
+            cursor = new GenerateSeriesTimestampStringRecordCursor(startFunc, endFunc, stepFunc);
+        }
+        cursor.of(executionContext);
+        return cursor;
+    }
+
+    @Override
+    public int getScanDirection() {
+        if (cursor != null && cursor.stride != 0) {
+            return cursor.stride > 0 ? SCAN_DIRECTION_FORWARD : SCAN_DIRECTION_BACKWARD;
+        }
+        return SCAN_DIRECTION_FORWARD;
+    }
+
+    @Override
+    public boolean recordCursorSupportsRandomAccess() {
+        if (cursor == null) {
+            return false;
+        } else {
+            return cursor.supportsRandomAccess();
+        }
+    }
+
+    private static class GenerateSeriesTimestampStringRecordCursor extends AbstractGenerateSeriesRecordCursor {
+        private final GenerateSeriesTimestampStringRecord recordA = new GenerateSeriesTimestampStringRecord();
+        private final GenerateSeriesTimestampStringRecord recordB = new GenerateSeriesTimestampStringRecord();
+        public int stride;
+        private TimestampAddFunctionFactory.LongAddIntFunction adder;
+        private long end;
+        private long start;
+        private char unit;
+
+        public GenerateSeriesTimestampStringRecordCursor(Function startFunc, Function endFunc, Function stepFunc) {
+            super(startFunc, endFunc, stepFunc);
+        }
+
+        @Override
+        public void calculateSize(SqlExecutionCircuitBreaker circuitBreaker, Counter counter) {
+            toTop();
+            while (hasNext()) {
+                counter.inc();
+            }
+        }
+
+        @Override
+        public Record getRecord() {
+            return recordA;
+        }
+
+        @Override
+        public Record getRecordB() {
+            if (supportsRandomAccess()) {
+                return recordB;
+            }
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public boolean hasNext() {
+            recordA.of(adder.add(recordA.curr, stride));
+            if (stride >= 0) {
+                return recordA.curr <= end;
+            } else {
+                return recordA.curr >= end;
+            }
+        }
+
+        public void of(SqlExecutionContext executionContext) throws SqlException {
+            super.of(executionContext);
+            this.start = startFunc.getTimestamp(null);
+            this.end = endFunc.getTimestamp(null);
+
+            final CharSequence str = stepFunc.getStrA(null);
+
+            if (str != null) {
```

Comment:
this looks poorly tested (given the relative complexity of the flow):
![Screenshot From 2025-06-23 10-31-28](https://github.com/user-attachments/assets/e5c2e298-faa4-486e-acd4-8590606bd374)


also, it looks to me `stride` can be left uninitialized in some cases?